roles/ceph-osd/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: install dependencies
+  apt: pkg=parted
+       state=present
+
+- name: write out bootstrap keyring
+  copy:
+    dest: "{{ hostvars[groups['mons'][0]]['keyrings']['source'] }}"
+    content: "{{ hostvars[groups['mons'][0]]['keyrings']['content'] | b64decode }}"
+
+- name: prepare osd disk(s)
+  command: "ceph-disk prepare {{ item.0 }} {{ item.1 }}"
+  with_together:
+    - ceph_osd.devices
+    - ceph_osd.raw_journal_devices
+  changed_when: false
+  failed_when: false
+
+- name: activate osd(s)
+  command: ceph-disk activate {{ item | regex_replace('^(\/dev\/cciss\/c[0-9]{1}d[0-9]{1})$', '\\1p') }}1
+  with_items: ceph_osd.devices
+  failed_when: false
```

Comment:
Generally changed would be masked if it's a command like task used to just gather a bit of info, rather than causing some action to happen. In this case, there is already a conditional on the task so the task should only trigger when desired. The task does result in actual change on the system so that should be indicated.

As for masking failure, that can be used for when just "trying" something, and it's acceptable that the trail fails. This doesn't come up very often, and should be avoided if possible. Blanket masking failure can lead to unexpected state later in the playbook.